app/migrations/synonyms/index.ts

```diff
@@ -0,0 +1,101 @@
+import { DynamoDBClient, QueryCommand } from '@aws-sdk/client-dynamodb'
+import { GetParameterCommand, SSMClient } from '@aws-sdk/client-ssm'
+import {
+  BatchWriteCommand,
+  DynamoDBDocumentClient,
+  paginateScan,
+} from '@aws-sdk/lib-dynamodb'
+import { slug } from 'github-slugger'
+import { orderBy, uniqBy } from 'lodash'
+
+import type { Circular } from '~/routes/circulars/circulars.lib'
+
+export interface WriteData {
+  eventId: string
+  initialDate: number
+}
+
+async function getTableNameFromSSM(dynamoTableName: string) {
+  const ssmClient = new SSMClient({ region: 'us-east-1' })
+  const command = new GetParameterCommand({ Name: dynamoTableName })
+  const response = await ssmClient.send(command)
+  return response.Parameter?.Value
+}
+
+export async function backfillSynonyms() {
+  const startTime = new Date()
+  console.log('Starting SYNONYM backfill...', startTime)
+  const eventsRun = new Set()
+  const dynamoCircularsTableName = '/RemixGcnProduction/tables/circulars'
+  const circularTableName = await getTableNameFromSSM(dynamoCircularsTableName)
+  const dynamoSynonymsTableName = '/RemixGcnProduction/tables/synonyms'
+  const TableName = await getTableNameFromSSM(dynamoSynonymsTableName)
+  const client = new DynamoDBClient({ region: 'us-east-1' })
+  const docClient = DynamoDBDocumentClient.from(client)
+  const pages = paginateScan(
+    { client: docClient },
+    { TableName: circularTableName, Limit: 25 }
+  )
+
+  for await (const page of pages) {
+    const writes = [] as WriteData[]
+    const circulars = page.Items as Circular[]
+
+    for (const circular of circulars) {
+      if (circular.eventId && !eventsRun.has(circular.eventId)) {
+        const command = new QueryCommand({
+          TableName: circularTableName,
+          IndexName: 'circularsByEventId',
+          KeyConditionExpression: 'eventId = :eventId',
+          ExpressionAttributeValues: {
+            ':eventId': { S: circular.eventId },
+          },
+        })
+
+        const response = await client.send(command)
+        const items = (response.Items || []).map((item) => ({
+          circularId: item.circularId?.S,
+          eventId: item.eventId?.S,
+          createdOn: item.createdOn?.N,
+        }))
```

Comment:
Use `docClient` so that you don't have to marshal and unmarshal types yourself.

Reply:
If I remove the type, you can see that it is now storing an AttributeValue
<img width="546" alt="Screenshot 2025-03-24 at 3 43 07 PM" src="https://github.com/user-attachments/assets/839982b6-c731-41ca-ac71-bd87a413f3f6" />

Reply:
That's because you imported `QueryCommand` from `@aws-sdk/client-dynamodb`. You need to import it from `@aws-sdk/lib-dynamodb`.